### PR TITLE
Clustering tech liz

### PR DESCRIPTION
**Pull request by lizgzil**
This includes:
- A script to cluster all the grants, and all the tech grants.
- Also a notebook to show the plots and some other analysis about the clustering.
- The html for the bokeh cluster plots (hope these work ok)

The clusters created here are stored [here](https://wellcomecloud.sharepoint.com/sites/DataLabs/Machine%20Learning/Forms/AllItems.aspx?newTargetListUrl=%2Fsites%2FDataLabs%2FMachine%20Learning&viewpath=%2Fsites%2FDataLabs%2FMachine%20Learning%2FForms%2FAllItems%2Easpx&viewid=a99f51bc%2D20ec%2D4b35%2D9090%2D4eedc2450aec&id=%2Fsites%2FDataLabs%2FMachine%20Learning%2FNutrition%20Labels%2Fmodels%2Fclustering)

The type of thing I looked at in the notebook was:

- How many in each?
- What proportion of tech grants in each?
- Plot clusters in all grant space
- Plot tech grants in all grant space
- Plot most and least techy grants in all grant space
- Plot tech clusters in tech grant space
- Keywords in highest and lowest techy grants
- Keywords in tech grants

To do: 
Biggest in terms of money clusters

### CODE REVIEW

File: nutrition_labels/cluster_tech_grants.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    grant_data = clean_grants_data(grant_data)
+    grants_text = grant_data[['Title', 'Grant Programme:Title', 'Description']].agg(
+                '. '.join, axis=1
+                ).tolist()
+    grant_data['Grant Text'] = grants_text
+    tech_grant_ids = tech_grants['Internal ID'].tolist()
+    grant_data['Tech grant?'] = grant_data['Internal ID'].isin(tech_grant_ids)
+    grant_data.sort_values(by=['Tech grant?'], ascending=False, inplace=True)
+    grant_data.drop_duplicates(subset=['Internal ID 6 digit'], inplace=True)
+
```
Comment (nonieA) [inline nutrition_labels/cluster_tech_grants.py:61]: I dont think you made the column 'Internal ID 6 digit' unless its in your version of wellcome-grants-awarded-2005-2019.csv

**Review by nonieA — APPROVED** [review]
Looks good

File: nutrition_labels/cluster_tech_grants.py (hunk 1 above)
Comment (lizgzil) [reply nutrition_labels/cluster_tech_grants.py:61]: it should be created when `clean_grants_data` is called

Comment (nonieA) [reply nutrition_labels/cluster_tech_grants.py:61]: yep completely my bad I must have skipped that line by accident when I was going through the code

**Pull request merged**